core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -567,7 +567,11 @@ public boolean decide(String workflowId) throws Exception {
             if (!tasksToBeRequeued.isEmpty()) {
                 addTaskToQueue(tasksToBeRequeued);
             }
-            workflow.getTasks().addAll(tasksToBeScheduled);
+
+            if(!workflow.getTasks().contains(tasksToBeScheduled))
+            {
```

Comment:
Good catch!
I think a cleaner way would be 
```java
tasksToBeScheduled.stream()
                    .filter(task -> !workflow.getTasks().contains(task))
                    .forEach(workflow.getTasks()::add);
```

Reply:
@huangyiminghappy please make the change according to Anoop's recommendation